src/mo_frontend/typing.ml

```diff
@@ -1874,13 +1874,15 @@ and infer_pat' env pat : T.typ * Scope.val_env =
         display_typ_expand t2;
     if T.Env.keys ve1 <> T.Env.keys ve2 then
       error env pat.at "M0184" "different set of bindings in pattern alternatives";
-    let check_same_bind_type bind t1 t2 = if not (T.eq t1 t2) then
-      error env pat.at "M0185" "types for alternative pattern variables %s differ%a%a"
+    let warn_lossy_bind_type bind t1 t2 = if not (T.eq t1 t2) then
+      let common = T.lub t1 t2 in
+      if T.(not (eq t1 common || eq common t2)) then
```

Comment:
I think elsewhere we use predicate `consistent` to  detect information. Maybe follow that pattern, for, err, consistency.

Reply:
If I read the `inconsistent` predicate correctly, it returns `true` when the `lub` is pushed right up to `Any`. I try to warn when both are smaller than the `lub`. So I changed the wording only.